rebuild/src/addon.h

```diff
@@ -0,0 +1,177 @@
+            .ThrowAsJavaScriptException();                                     \
+        m_has_error = true;                                                    \
+        return;                                                                \
+    }                                                                          \
+    Napi::Uint8Array arr_name =                                                \
+        arr_name##_array.As<Napi::TypedArrayOf<uint8_t>>();
+
+class BlstTsAddon;
+
+typedef enum { Affine, Jacobian } CoordType;
+
+/**
+ * Checks a byte array to see if it is all zeros. Can pass start byte for the
+ * cases where the first byte is the tag (infinity point and
+ * compress/uncompressed).
+ *
+ * @param data uint8_t*
+ * @param start_byte size_t
+ * @param byte_length size_t
+ *
+ * @return bool
+ */
+bool is_zero_bytes(
+    const uint8_t *data, const size_t start_byte, const size_t byte_length);
+
+/**
+ * Checks if a byte array is a valid length. If not, sets the error message and
+ * returns false.  If valid returns true for use in conditional statements.
+ *
+ * @param[out] error_out &std::string - error message to set if invalid
+ * @param[in] byte_length size_t - length of the byte array to validate
+ * @param[in] length1 size_t - first valid length
+ * @param[in] length2 size_t - second valid length (optional)
+ *
+ * @return bool
+ */
+bool is_valid_length(
+    std::string &error_out,
+    size_t byte_length,
+    size_t length1,
+    size_t length2 = 0);
+
+/**
+ * Circular dependency if these are moved up to the top of the file.
+ */
+#include "functions.h"
+#include "public_key.h"
+#include "secret_key.h"
+#include "signature.h"
+
+/**
+ * BlstTsAddon is the main entry point for the library. It is responsible
+ * for initialization and holding global values.
+ */
+class BlstTsAddon : public Napi::Addon<BlstTsAddon> {
+   public:
+    std::string _dst;
+    std::string _blst_error_strings[8];
+    Napi::FunctionReference _secret_key_ctr;
+    Napi::FunctionReference _public_key_ctr;
+    Napi::FunctionReference _signature_ctr;
+
+    /**
+     * BlstTsAddon::BlstTsAddon constructor used by Node.js to create an
+     * instance of the addon.
+     *
+     * @param env Napi::Env
+     * @param exports Napi::Object
+     *
+     * @return BlstTsAddon
+     *
+     * @throws Napi::Error
+     */
+    BlstTsAddon(Napi::Env env, Napi::Object exports);
+
+    /**
+     * References are by default non-copyable and non-movable. This is just
+     * to make it explicit that it's not allowed to be copied or moved.
+     */
+    BlstTsAddon(BlstTsAddon &&source) = delete;
+    BlstTsAddon(const BlstTsAddon &source) = delete;
+    BlstTsAddon &operator=(BlstTsAddon &&source) = delete;
+    BlstTsAddon &operator=(const BlstTsAddon &source) = delete;
+
+    /**
+     * Converts a blst error to an error string
+     */
+    std::string GetBlstErrorString(const blst::BLST_ERROR &err);
```

Comment:
It could be a free function, given that it seems to be a helper function, not really operating on the instance.

Reply:
It is not possible to use static variables due to the nature of how the `dylib` is loaded by node.  That is why the error strings are instance values of the class.  And by extension its why the function is as well so it has access to the data.